graphium/fingerprinting/data.py

```diff
@@ -0,0 +1,216 @@
+from typing import Any, List, Dict, Literal, Union
+
+import os
+
+import torch
+
+import pandas as pd
+import numpy as np
+
+from pytorch_lightning import LightningDataModule
+
+from torch.utils.data import Dataset, DataLoader
+
+from graphium.data.datamodule import MultitaskFromSmilesDataModule, ADMETBenchmarkDataModule
+from graphium.trainer.predictor import PredictorModule
+from graphium.fingerprinting.fingerprinter import Fingerprinter
+
+
+class FingerprintDataset(Dataset):
+    def __init__(
+            self,
+            smiles: List[str],
+            labels: torch.Tensor,
+            fingerprints: Dict[str, torch.Tensor],
+    ):
+        self.smiles = smiles
+        self.labels = labels
+        self.fingerprints = fingerprints
+
+    def __len__(self):
+        return len(self.smiles)
+
+    def __getitem__(self, index):
+        fp_list = []
+        for val in self.fingerprints.values():
+            fp_list.append(val[index])
+        return fp_list, self.labels[index]
+
+
+class FingerprintDatamodule(LightningDataModule):
```

Comment:
Why not inherit from the MultitaskDatamodule? Is it because you sometimes need the "ADMETDatamodule"?

Reply:
Yes, `MultitaskDatamodule` or `TDCDatamodule` are used depending on the data source. There could be an argument for making `FingerprintDatamodule.prepare_data` a function (`create_fps`) of the `BaseDataModule` but this would require major changes. I suggest keeping it in mind for future improvements.